### PR TITLE
HPCC4J-532 Jirabot python script fixes

### PR DESCRIPTION
**Pull request by jpmcmu**
- Fixed status mapping
- Improved code flow
- Improved issue commenting
- Moved issue update to separate function

Signed-off-by: James McMullan James.McMullan@lexisnexis.com

## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [X] I have created a corresponding JIRA ticket for this submission
- [X] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [ ] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by jpmcmu** [comment]
@rpastrana please review

**Comment by jpmcmu** [comment]
@rpastrana Pushed up code review changes, please review

File: .github/workflows/JirabotMerge.yml [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+            def buildVersionString(version):
+                major, minor, point = map(int, version)
+                return f"{major}.{minor}.{point}"
+
+            def generateFixVersionList(jira, branchName):
+
+                latestVersion = getTagVersionForCmd("git tag --list 'hpcc4j_*-release' --sort=-v:refname | head -n 1")
+
+                # If we are merging into master we assume it is going into the next minor release
```
Comment (rpastrana) [inline .github/workflows/JirabotMerge.yml:83]: As discussed, there's a scenario where master is reserved for next major, I wonder if we can infer this situation if we find a branch for the next minor but no corresponding tags...
master version 10.1.0
latest tag 10.4.10
branch 10.6.x exists

**Comment by rpastrana** [comment]
@jpmcmu squash this one

**Comment by jpmcmu** [comment]
@rpastrana squashed

**Pull request merged**